app/common/vecnest_util.c

```diff
@@ -0,0 +1,73 @@
+
+#include <petsc.h>
+#include <petscvec.h>
+#include <petsc/private/vecimpl.h>
```

Comment:
Is this header part of the petsc installation? (And why is it needed here?)

Reply:
> 

Yes it is. It is needed as in the last function I need to set set some function points (`x->ops->`). There might be an API available to set those but I don't think so. There is for Mat and PC (for use with the Shell's) but not for Vec (to my knowledge).